backend/ASTU-backend-group-1/delivery/routers/BlogRouter.go

```diff
@@ -0,0 +1,70 @@
+package router
+
+import (
+	"astu-backend-g1/delivery/controllers"
+
+	"github.com/gin-gonic/gin"
+)
+
+type BlogRoute struct {
+	usecase controllers.BlogController
+	authController controllers.GeneralAuthorizationController
+}
+
+func NewBlogRoute(usecase controllers.BlogController,authcontroller controllers.GeneralAuthorizationController) *BlogRoute {
+	return &BlogRoute{usecase: usecase,authController: authcontroller}
+}
+
+func (gr *BlogRoute) GinBlogRouter() {
+	router := gin.Default()
+	blogRouter := router.Group("/blog")
+	{
+		// INFO:PASSED
+		blogRouter.POST("/new", gr.authController.UserMiddlewareGin(), gr.usecase.HandleCreateBlog)
+		blogRouter.GET("/all", gr.authController.UserMiddlewareGin(), gr.usecase.HandleGetAllBlogs)
+		blogRouter.GET("/popular", gr.authController.UserMiddlewareGin(), gr.usecase.HandleGetPopularBlog)
+		blogRouter.GET("/filter", gr.authController.UserMiddlewareGin(), gr.usecase.HandleFilterBlogs)
+		blogRouter.PATCH("/:blogId", gr.authController.UserMiddlewareGin(), gr.usecase.HandleUpdate)
+		blogRouter.DELETE("/:blogId", gr.authController.AdminMiddlewareGin(), gr.usecase.HandleDelete)
+		blogRouter.PATCH("/interact/:type", gr.authController.UserMiddlewareGin(), gr.usecase.HandleBlogLikeOrDislike)
+
+	}
+	commentRouter := router.Group("/comment")
+	{
+		// TODO: check if there is a blog with such id
+		commentRouter.PATCH("/new", gr.authController.UserMiddlewareGin(), gr.usecase.HandleCommentOnBlog)
```

Comment:
Change the method to post method instead of patch
and also make the /comment to /comments and remove the /new